core/frontend/src/extension/Extension.ts

```diff
@@ -0,0 +1,58 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Extensions
+ */
+
+/**
+ * @beta
+ */
+export enum ActivationEvent {
+  onStartup = "onStartup",
+}
+
+/**
+ * @beta
+ */
+export type ResolveFunc = () => Promise<any>;
+
+/** Defines the format of an Extension manifest
+ * @beta
+*/
+export interface ExtensionManifest {
+  /** The extension name */
+  readonly name: string;
+  /** The extension display name */
+  readonly displayName?: string;
+  /** The extension version */
+  readonly version: string;
```

Comment:
Does this follow semantic versioning, or otherwise expect a specifically-formatted string?

Reply:
this will be the version from package.json in most cases, so it "should" be semver but we haven't established that yet. I think its really up to the extension to just ensure versions are unique